src/intelligence_layer/core/model.py

```diff
@@ -585,6 +523,57 @@ def generate_chat(
 
         return self.generate(prompt_item.text, tracer)
 
+    def echo_chat(
+        self,
+        messages: list[Message],
+        response_prefix: str | None,
+        expected_completion: str,
+        tracer: Tracer,
+    ) -> Sequence[tuple[Any, float | None]]:
+        prompt = self.to_chat_prompt(messages, response_prefix)
+        prompt_item = prompt.items[0]
+        assert isinstance(prompt_item, Text)
+
+        return self.echo(prompt_item.text, expected_completion, tracer)
+
+
+LLAMA_3_PROMPT_TEMPLATE = PromptTemplate(
+    """<|begin_of_text|>{% for message in messages %}<|start_header_id|>{{message.role}}<|end_header_id|>
+
+{% promptrange instruction %}{{message.content}}{% endpromptrange %}<|eot_id|>{% endfor %}<|start_header_id|>assistant<|end_header_id|>
+
+{% if response_prefix %}{{response_prefix}}{% endif %}"""
+)
+
+
+class Pharia1ChatModel(AlephAlphaChatModel):
+    """Chat model to be used for any `"Pharia-1-LLM-*` model.
+
+    Args:
+        name: The name of a valid Pharia-1 model.
+            Defaults to `Pharia-1-LLM-7B-control`
+        client: Aleph Alpha client instance for running model related API calls.
+            Defaults to :class:`LimitedConcurrencyClient`
+    """
+
+    CHAT_PROMPT_TEMPLATE = LLAMA_3_PROMPT_TEMPLATE
```

Comment:
the model uses the llama3 template, which has `<|eot_id|>`, but its eot token is `"<|endoftext|>"`. Is this correct?

Reply:
Yes, it's a bit weird/confusing, but correct

In the input, you can use both eot_id and endoftext, but the model will only ever produce endoftext